shared/pub/metadata.js

```diff
@@ -0,0 +1,54 @@
+export const getPDFDownloads = (pub) => {
+	const downloads = pub.downloads;
+	const exports = pub.activeBranch.exports;
+	if (downloads && downloads.length > 0) {
+		return downloads.filter((download) => {
+			if (download.url.substring(download.url.length - 4) === '.pdf') {
+				return download;
+			}
+			return false;
+		});
+	}
+	if (exports && exports.length > 0) {
+		return exports.filter((exportFile) => {
+			if (exportFile.format === 'pdf') {
+				return exportFile;
+			}
+			return false;
+		});
+	}
+	return false;
+};
+
+export const getTextAbstract = (content) => {
```

Comment:
Let me suggest making this a bit more readable by destructing the document a bit and adding clarifying names:
```js
// Names like 'doc' or 'docJson' are conventional in our codebase to suggest
// that the variable is a Prosemirror doc
export const getTextAbstract = docJson => {
	const {content} = docJson;
	const [firstChild, secondChild] = content;
	// See whether the first child is the header we expect
	const firstChildIsAbstractHeader =
		firstChild &&
		firstChild.type === 'heading' &&
		firstChild.level === 1 &&
		firstChild.attrs.id === 'abstract';
	if (firstChildIsAbstractHeader && secondChild) {
		// Start extracting content from the second paragraph
		const {content: abstractContent} = secondChild;
		if (abstractContent) {
			abstractContent.forEach(/* ... */);
		}
	}
}
```

Reply:
Thank you. This is the type of code I would like to learn to write. I'm real rusty on destructuring and never think of stuff like creating named bools out of various properties.